### DISCUSSION TITLE
potree

### DISCUSSION THREAD

**Discussion by radoslavirha** [visgl/deck.gl#5690] [Q&A] [answered]
Hello is it possible to visualise potree point clouds? Thanks

**Comment by kylebarron** [comment, marked as answer]
I think there's been discussion about loading potree data in loaders.gl https://github.com/visgl/loaders.gl/issues/257 and https://github.com/visgl/loaders.gl/issues/212

**Reply by ibgreen** [reply]
We do have an experimental [`@loaders.gl/potree` module](https://github.com/visgl/loaders.gl/tree/master/modules/potree) that is intended to load and "normalize" potree data so that it can work with the deck.gl `Tile3DLayer`.

I seem to recall that there was a problem in the past that potree data did not contain geospatial reference point but this may have been addressed in latest version of potree. Alternatively it could be supplied as a parameter by the application.

It would likely not be hard to get everything working, just a matter of finding some interested contributor to help complete the effort.

**Reply by radoslavirha** [reply]
It does not work now, it's missing refine (refine is in Cesium 3d tiles I think). Potree data is .las converted by Current potreeConverter

```
const potree = await load(`${this.$configuration.api}${layer.path[0]}/metadata.json`);
const tileset = new Tileset3D(potree);
```

![image](https://user-images.githubusercontent.com/23722745/116078043-713c1400-a696-11eb-87e1-2500ee248e3d.png)

I also tried some urls from Potree examples (v1.7), but without success. 

![image](https://user-images.githubusercontent.com/23722745/116078548-0b9c5780-a697-11eb-9366-76dec68fcad5.png)

**Comment by ibgreen** [comment]
Understood. You could just do a one line change / PR and update the Tileset3D code to apply a default `refinement` (or set the field in the potree loader). Of course, there is no guarantee we don't hit additional issues. Would be happy to help you work through the issues as they come up. Depends on your appetite for getting your hands dirty...

**Reply by radoslavirha** [reply]
Hello,

I've tried to add refinement, but it won't be enough I think. My colleague sent me potree 2 format and it's different from older format. The major difference is in the file structure. Potree 2 is one huge `.bin` file loaded in chunks (inspected in potree viewer with potree 2 data), potree 1.7 is much more similar to cesium 3d tiles with bunch of smaller `.bin` files. So before we can help you, we need to decide which format is best for point clouds. What is your preference?

| format        | pros           | cons  |
| ------------- |:-------------:|:-----|
| cesium 3d point clouds | deck.gl support | conversion from .las/.laz  |
|  |  | FME converter is not good enough - poor LOD  |
|  |  | ION converter is good, but supports only few input formats  |
|  |  | cesium 3d point clouds from ENTWINE converter seems to be less performant (tested on colored point clouds) ... maybe deck.gl issue ... will test it more  |
| potree | easy conversion | not supported in deck.gl |
| .las/.laz | no conversion | max 100mb file |

We have huge georeferenced point clouds in .las/.laz format. Maybe we can convert it to `.ply` too and test it [https://loaders.gl/examples/pointcloud](url)? Can you please advise us what will be the best choice to visualise huge point clouds?

Thanks :)

**Reply by ibgreen** [reply]
Thanks for the update on potree 2 format. Yes it does indeed look like potree 2 requires supporting HTTP range requests to load out the data for each tile from a single resource (the data is still tiled, just needs to be loaded from ranges in a big file, rather than from individual files). Range request loading is not supported by the related loaders.gl code at this stage, though likely not too hard to add for someone making PRs.

I am not sure PLY is smaller than LAZ as LAZ is a compressed binary format (where as LAS is uncompressed per my understanding), and PLY is uncompressed. It depends to a certain extent on precision of positions (I think LAS supports 64 bits which if used obviously takes more space) and how many additional data columns you have per point "row". 

Another compressed point cloud format is Draco3D which is well supported by deck.gl and loaders.gl.

The main development focus for tiled 3d data in loaders.gl right now is I3S but unfortunately we have not yet implemented support for I3S point clouds, and I am not sure there are point cloud tiling tools that would help you.

It is unfortunate that your input files are not supported by ION. Another approach could be to add Writers for point cloud formats in loaders.gl so that you could use loaders.gl to convert to a format supported by Cesium ION.

There should be an experimental C++ point cloud tiler (generating 3D tiles) in loaders.gl `cpp` folder but it has not been maintained not sure if it is worth looking at right now.
